astrowidgets/glupyter.py

```diff
@@ -0,0 +1,400 @@
+            Memory mapping. See `astropy.io.fits.open`.
+
+        """
+        with fits.open(filename, memmap=memmap) as pf:
+            data = _new_data_from_hdulist(pf, numhdu)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def load_nddata(self, nddata, label=None):
+        """Load a `~astropy.nddata.NDData` object into the viewer.
+
+        .. todo:: Add flag/masking support, etc.
+
+        Parameters
+        ----------
+        nddata : `~astropy.nddata.NDData`
+            ``NDData`` with image data and WCS.
+
+        label : str or `None`
+            Glue label name for the data.
+
+        """
+        if label is None:
+            label = _gen_random_label()
+
+        data = Data(label=label)
+
+        if nddata.wcs is not None:
+            coords = coordinates_from_wcs(nddata.wcs)
+            data.coords = coords
+
+        data.add_component(component=nddata.data, label=label)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def load_array(self, arr, label=None):
+        """Load a 2D array into the viewer.
+
+        .. note:: Use :meth:`load_nddata` for WCS support.
+
+        Parameters
+        ----------
+        arr : array-like
+            2D array.
+
+        label : str or `None`
+            Glue label name for the data.
+
+        """
+        if label is None:
+            label = _gen_random_label()
+
+        data = Data(label=label)
+        data.add_component(component=arr, label=label)
+
+        viewer = self._viewer
+        data = validate_data_argument(self._app.data_collection, data)
+
+        # NOTE: Changing state via API does not work properly if we loaded
+        # multiple data into the same widget in the same session.
+        self._app.data_collection.clear()
+
+        self._app.add_datasets(data)
+        viewer.add_data(data)
+        viewer.state.reference_data = data  # Bring it to focus
+        self._link_image_to_cb()
+
+    def center_on(self, point):
+        raise NotImplementedError  # FIXME
+
+    def offset_to(self, dx, dy, skycoord_offset=False):
+        raise NotImplementedError  # FIXME
+
+    @property
+    def zoom_level(self):
```

Comment:
you could get x_min/x_max/y_min/y_max as in center_on and compute the fraction of the image shown and return that as zoom level?

Reply:
I am surprised that zoom info isn't already built into Glue. You have a zoom/pan button after all.